hopsworks-api/src/main/java/io/hops/hopsworks/api/kafka/KafkaResource.java

```diff
@@ -324,117 +335,287 @@ public Response updateTopicAcls(@Context UriInfo uriInfo, @PathParam("topic") St
     return Response.ok(uri).entity(aclDto).build();
   }
 
-  //validate the new schema
+  @ApiOperation(value = "Get schema by its id.")
+  @GET
+  @Path("/schemas/ids/{id}")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  @JWTRequired(acceptedTokens={Audience.API}, allowedUserRoles={"HOPS_ADMIN", "HOPS_USER"})
+  public Response getSchema(@PathParam("id") Integer id) {
+    try {
+      SubjectDto dto = schemasController.findSchemaById(project, id);
+      return Response.ok().entity(dto).build();
+    } catch (SchemaException e) {
```

Comment:
This needs to be done in the exception mapper.

Reply:
the exceptions follow a different format than all the other in Hopsworks (https://docs.confluent.io/5.3.0/schema-registry/develop/api.html#errors). is exception mapper configurable to introduce a different format?

Reply:
Yes, you can add any mapping

Reply:
Agreed to leave it like this only for the Schema Registry endpoints